### ISSUE TITLE
What would I need to do to config this repo to work for me?

### ISSUE THREAD

**Issue by cowlicker12** [richardle17/SellerPortalDatabase#1]
I see the last note in the ReadMe that says that this repo is intended to keep track of your own person ecommerce inventory, but is there any way to use this for myself?

**Comment by RichyLe69** [comment]
Yes you can you would need to create a tcgplayer.com seller's account and log into their SellerPortal https://store.tcgplayer.com/admin/Account/Settings.

Then install selenium driver on your own computer, update the executable path as seen in the main.py (because currently it uses my own pc's path) 

I currently use list.yaml to control what list of cards I scrape, you can change it to whichever ones you want. 

Then for YOUR cards, you'd just put in your cards in this format:

Book of moon cp1:
  url: 26057
  edition: 'Near Mint'
  qty: 1

where 'url' is the id number of the cards in the seller portal inventory link
example: https://store.tcgplayer.com/admin/product/manage/26057
get that number at the end.

Try scraping 1 card, then if that works, u just keep adding more cards.

**Comment by cowlicker12** [comment]
How do I slim everything down to work with only 1 list? I tried commenting
out everything in list.yaml and then creating a path to only my
particularCardList.yml, but it seems like there is more to do. Right now,
it seems to run but in a wonky way. It opens Chrome on the page for
ultimate rare Judgment Dragon, but my intention was to scrape the prices
for Dogmatika Ecclesia, the Virtuous.

**Comment by RichyLe69** [comment]
> 

Yes the judgment dragon is just the "default" page I set for myself. In the console terminal it should say "enter any key to begin" or something along those lines. I just press enter and it should automatically go through your list.yaml as long as you put the correct ID number for that card

**Comment by cowlicker12** [comment]
Ah ok, the Judment Dragon page makes sense. I might change this to Saggi
the Dark Clown for the lolz.

I'm getting an error about a missing positional argument. The only changes
I made to your entire repo is to remove the dependency on executable path
for the webdriver because at some point I read something saying that using
executable path is deprecated. I left everything else the same, including
your card lists. Could it be that we have different installations of
PyYAML? I installed mine using the following commands:

pip install ruamel.yaml

pip install PyYAML

[image: image.png]

**Comment by cowlicker12** [comment]
Actually never mind - I found a SO post that says to use yaml.safe_load()
instead of yaml.load() and that cleared this error:
https://stackoverflow.com/questions/69564817/typeerror-load-missing-1-required-positional-argument-loader-in-google-col

I see the browser now iterating through a bunch of webpages. Below is the
output. Any chance that your repo can output this to a file? My goal is to
insert prices for a particular list of cards into a SQL Server table, so
either a text file or an Excel file would work.

Also, have you had any issues with TCGPlayer getting angry that you are
pinging their website so quickly? I'd like to track prices for thousands of
cards, so I'm worried they might block my IP.

[image: image.png]

**Comment by RichyLe69** [comment]
yeah all that was on my todo list but haven't had the chance to explore that stuff. at the moment It only outputs the plain text tables with the info it scrapes. You could easily pass the 3 card value data into something else and make that into something more compact

as for issues with tcgplayer, i've done hundreds of cards every few days for over a year, havn't had any issues so far. I have not tried thousands of cards though. 

Actually my old version of this project (TcgPlayerScraper) which used the regular card pages (not the seller portal page) I remember at one point they knew that I was automating and had some sort of warning page, but I haven't gotten such an error when using the seller portal page.

**Comment by cowlicker12** [comment]
Hello again,

I stumbled onto a realization while working on this that I thought you
should know about. The lowest price that you see in the seller portal is
sometimes nowhere near the lowest price on the regular card pages. I think
it might be because TCGPlayer wants to show customers the lowest prices *but
only for the highest rated sellers*. Here's an example - just the other day
I was seeing that the lowest price for a card was about $1.25 in the seller
portal. When I went over to the regular card page, the lowest I saw in the
list of cards (for the given edition and condition) was about $3.00. This
seemed strange until I noticed that the whole first page of listings on the
regular card page were all highly rated vendors with lots of
sales/feedback. And the reason this is important is because whenever I
trade people for cards in person, we go off of the top 5 listings that we
see on the regular card pages - these are the prices that the consumers
consider the "true" price of a card.

Some food for thought.

**Comment by cowlicker12** [comment]
Hey,

Quick question - is it possible (and relatively easy) to make this repo
read the card list from a CSV file instead of a YAML file?

Thanks,
Aaron

**Comment by RichyLe69** [comment]
Should be pretty simple, just read the data from a csv where each row would have a format somewhat like this: name,url,editioncondition,qty.

Get those values into some sort of json format, and feed it into the scraper
